plugins/net_plugin/net_plugin.cpp

```diff
@@ -3598,7 +3598,8 @@ namespace eosio {
          hello.token = sha256();
       hello.p2p_address = my_impl->p2p_address;
       if( is_transactions_only_connection() ) hello.p2p_address += ":trx";
-      if( is_blocks_only_connection() ) hello.p2p_address += ":blk";
+      // if we are not accepting transactions tell peer we are blocks only
+      if( is_blocks_only_connection() || !my_impl->p2p_accept_transactions ) hello.p2p_address += ":blk";
```

Comment:
Can `!my_impl->p2p_accept_transactions` be folded inside `is_blocks_only_connection()`?

Reply:
No, as other places `is_blocks_only_connection()` is called we don't want it to be true. For example `is_blocks_only_connection()` is called in the loop to send out trxs. If it was true we would not send out a trx to our peer. We want to send trxs out, just not allow transactions to flow in.

Reply:
Thanks. Maybe you can add a comment here.